Lib/slapdtest/_slapdtest.py

```diff
@@ -223,13 +251,15 @@ def __init__(self):
         self.clientkey = os.path.join(HERE, 'certs/client.key')
 
     def _check_requirements(self):
-        binaries = [
-            self.PATH_LDAPADD, self.PATH_LDAPMODIFY, self.PATH_LDAPWHOAMI,
-            self.PATH_SLAPD, self.PATH_SLAPTEST
-        ]
-        for binary in binaries:
-            if not os.path.isfile(binary):
-                raise ValueError('Binary {} is missing.'.format(binary))
+        for name in dir(self):
+            if not name.startswith('PATH_'):
+                continue
+            value = getattr(self, name)
+            if value is None or not os.path.isfile(value):
+                cmd = name[5:].lower()
+                raise ValueError(
+                    "Command '{}' not found in PATH".format(cmd)
+                )
```

Comment:
So … commands are defined by being prefixed by `PATH_`?
That looks like avoidable magic. I think it would be better to have the explicit list and just add a comment to remind people about synchronization, or to turn `_which` into something that registers the command as well as finding it.

Reply:
The old list of file names doesn't result in good UX because ``_which`` returns ``None`` when it can't find a command. An error message like ``command None not found`` isn't very helpful. What do you suggest?